app/assets/javascripts/discourse/tests/integration/components/user-menu/messages-list-test.js

```diff
@@ -5,14 +5,128 @@ import { render, settled } from "@ember/test-helpers";
 import { NOTIFICATION_TYPES } from "discourse/tests/fixtures/concerns/notification-types";
 import { hbs } from "ember-cli-htmlbars";
 import pretender, { response } from "discourse/tests/helpers/create-pretender";
+import { cloneJSON, deepMerge } from "discourse-common/lib/object";
+import UserMenuFixtures from "discourse/tests/fixtures/user-menu";
 import I18n from "I18n";
 
+function getMessage(overrides = {}) {
+  return deepMerge(
+    {
+      id: 8092,
+      title: "Test ToPic 4422",
+      fancy_title: "Test topic 4422",
+      slug: "test-topic-4422",
+      posts_count: 1,
+      reply_count: 0,
+      highest_post_number: 2,
+      image_url: null,
+      created_at: "2019-07-26T01:29:24.008Z",
+      last_posted_at: "2019-07-26T01:29:24.177Z",
+      bumped: true,
+      bumped_at: "2019-07-26T01:29:24.177Z",
+      unseen: false,
+      last_read_post_number: 2,
+      unread_posts: 0,
+      pinned: false,
+      unpinned: null,
+      visible: true,
+      closed: false,
+      archived: false,
+      notification_level: 3,
+      bookmarked: false,
+      bookmarks: [],
+      liked: false,
+      views: 5,
+      like_count: 0,
+      has_summary: false,
+      archetype: "private_message",
+      last_poster_username: "mixtape",
+      category_id: null,
+      pinned_globally: false,
+      featured_link: null,
+      posters: [
+        {
+          extras: "latest single",
+          description: "Original Poster, Most Recent Poster",
+          user_id: 13,
+          primary_group_id: null,
+        },
+      ],
+      participants: [
+        {
+          extras: "latest",
+          description: null,
+          user_id: 13,
+          primary_group_id: null,
+        },
+      ],
+    },
+    overrides
+  );
+}
+
+function getGroupMessageSummaryNotification(overrides = {}) {
+  return deepMerge(
+    {
+      id: 9492,
+      user_id: 1,
+      notification_type: 16,
+      read: true,
+      high_priority: false,
+      created_at: "2022-08-05T17:27:24.873Z",
+      post_number: null,
+      topic_id: null,
+      fancy_title: null,
+      slug: null,
+      data: {
+        group_id: 1,
+        group_name: "jokers",
+        inbox_count: 4,
+        username: "joker.leader",
+      },
+    },
+    overrides
+  );
+}
+
 module("Integration | Component | user-menu | messages-list", function (hooks) {
   setupRenderingTest(hooks);
 
   const template = hbs`<UserMenu::MessagesList/>`;
 
-  test("renders notifications on top and messages on bottom", async function (assert) {
+  test("renders unread PM notifications first followed by messages and read group_message_summary notifications", async function (assert) {
```

Comment:
The title here and the assertion below has a mismatch of ordering. I think it'll be better if we can more intentionally call out the ordering as per the PR description in the tests here.